src/core/server/elasticsearch/legacy/cluster_client.ts

```diff
@@ -257,7 +260,8 @@ export class LegacyClusterClient implements ILegacyClusterClient {
     }
     const authHeaders = this.getAuthHeaders(request);
     const headers = ensureRawRequest(request).headers;
+    const requestIdHeaders = request instanceof KibanaRequest ? { 'x-opaque-id': request.id } : {};
 
-    return { ...headers, ...authHeaders };
+    return { ...headers, ...authHeaders, ...requestIdHeaders };
```

Comment:
Probably minor, but order inconsistency with the new client:

`{ ...requestHeaders, ...requestIdHeaders, ...authHeaders }`

vs

`{ ...headers, ...authHeaders, ...requestIdHeaders }`